image/image.go

```diff
@@ -380,6 +416,15 @@ func bootstrapToRootDir(tsto *ToolingStore, model *asserts.Model, opts *Options,
 			fmt.Fprintf(Stdout, "Fetching %s\n", snapName)
 		}
 
+		dlOpts.Channel = opts.Channel
+		if snapName == model.Kernel() && model.KernelTrack() != "" {
+			kch, err := makeKernelChannel(model.KernelTrack(), opts.Channel)
+			if err != nil {
+				return err
+			}
+			dlOpts.Channel = kch
```

Comment:
Same question here. This is overriding something that was provided elsewhere. Is it better to overwrite (as done now) or is better to conflict? If we're accepting a parameter from elsewhere that explicitly says the kernel should come from a different channel, conflicting might be the right thing to do.

Reply:
After we've read til the end, I think we understand that this is the channel for every snap.. is it?  If so, the logic here feels good, but the overall behavior feels bad. :) But that's a different issue we'll need to solve elsewhere.